js/entity_browser.modal.js

```diff
@@ -37,4 +37,60 @@
       });
     }
   };
+
+  Drupal.behaviors.fluidModal = {
+    attach: function (context) {
+
+      // on window resize run function
+      $(window).resize(function (context) {
+        fluidDialog();
+      });
+
+      // catch dialog if opened within a viewport smaller than the dialog width
+      // run function on all dialog opens
+      $(document).on("dialogopen", ".ui-dialog", function (event, ui) {
+        fluidDialog();
+      });
+    }
+  };
+
+  var fluidDialog = function fluidDialog() {
```

Comment:
Let's put this in "Drupal.entityBrowserModal.fluidDialog". See examples in entity_browser.common.js.